### ISSUE TITLE
New feature discussion - generate HTML report

### ISSUE THREAD

**Issue by ptrin** [OpenAssessItToolkit/openassessit#17]
I've created a rough POC of generating an HTML report from the Markdown:
https://github.com/ptrin/openassessit/commit/f6124a96b10e14b545a289df022b3f572af8f861
https://github.com/ptrin/openassessit_templates/commit/a02d0b35987dfa23c39b1d3b8ff5eca854737250

This is not ready for a PR for a few reasons:
- Hardcoded `--html` option in the run script
- The output filename should be properly suffixed with `.md` or `.html` depending on the `--html` option. For now it has to be manually changed.
- Hardcoded user template path to my own working dir

Please take a look and let me know how you think we should clean this up.

**Comment by joelhsmith** [comment]
Awesome thanks @ptrin I'll have time to try it out next week.

**Comment by joelhsmith** [comment]
Awesome work @ptrin 

Thanks for doing the hard part! I used your code to add the .html export functionality.

I used your main Markdown and TOC code as you had it.  But I moved it into its own file [to_html.py](openassessit/openassessit/to_html.py) file.  

https://github.com/OpenAssessItToolkit/openassessit/blob/943e79063479e93ced5450ad31a86f8127c1d086/openassessit/to_html.py#L67-L78

My design rationale for moving to be its own separate thing was that I think that people will edit the MD file often.  Making it a separate script will allow them to run it on demand as they make edits.  I did add it to the [run.sh](openassessit/run.sh ) and the Docker [entrypoint.sh]( openassessit/entrypoint.sh) so it will run by default, if people choose to use them.

to_html.py now duplicates a bit of code from markdown.py. I think I can refactor some stuff into utils.py to help make it DRY-er than I made it. 

I am using the compiled Pygment CSS like you did and put it in the [header html template](openassessit/openassessit/templates/to_html_header.html).  Only thing I changed here was instead of putting it in a separate file, I added it directly into a <style> tag in the <head> for simplicity. If it is in the head, we don’t need to copy any dependencies around.  Also, the template folder only includes template files by moving the CSS inside the HTML file.

Again, thanks for working out that stuff for me. You save me a lot of time and built something better than I would have.

Its not code complete, but I think we are feature complete.  Its a bug fixing phase now, like improving in the image capture, refactor to be cleaner overall, etc. Is there anything else we should add before calling it feature complete and tagging as an alpha release?

I added your name to the readme.md file as a contributor, i hope thats ok ;-)
